Source/WebCore/platform/graphics/cocoa/MediaPlayerPrivateWebM.mm

```diff
@@ -198,33 +198,68 @@ static bool isCopyDisplayedPixelBufferAvailable()
     return SourceBufferParserWebM::isContentTypeSupported(parameters.type);
 }
 
-void MediaPlayerPrivateWebM::load(const String& url)
+void MediaPlayerPrivateWebM::setPreload(MediaPlayer::Preload preload)
```

Comment:
This method is doing much more than setting preload, it is actually preloading if needed.
Maybe the name should be clarified (and yes the setPreload overriding method would call this new method).
Something like `doPreload()`

Reply:
setPreload is the name of the virtual method, I copied the logic and the code from the MediaPlayerPrivateAVFoundation

Reply:
I wouldn’t assume everything `MediaPlayerPrivateAVFoundation` does is correct